baremetal/metal3datatemplate_manager_test.go

```diff
@@ -444,6 +444,143 @@ var _ = Describe("Metal3DataTemplate manager", func() {
 		}),
 	)
 
+	type testCaseTemplateReference struct {
+		template1                  *infrav1.Metal3DataTemplate
+		template2                  *infrav1.Metal3DataTemplate
+		dataObject                 *infrav1.Metal3Data
+		dataClaim                  *infrav1.Metal3DataClaim
+		indexes                    map[int]string
+		expectError                bool
+		expectTemplateReference    bool
+		expectDataObjectAssociated bool
+	}
+
+	DescribeTable("Test Template Reference",
+		func(tc testCaseTemplateReference) {
+			objects := []runtime.Object{}
+			objects = append(objects, tc.dataClaim)
+			c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), objects...)
+			templateMgr, err := NewDataTemplateManager(c, tc.template2,
+				klogr.New(),
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			_, err = templateMgr.createData(context.TODO(), tc.dataClaim, tc.indexes)
+			if tc.expectError {
+				Expect(err).To(HaveOccurred())
+			} else {
+				Expect(err).NotTo(HaveOccurred())
+			}
+
```

Comment:
What about this?
```suggestion
			   dataObjects := infrav1.Metal3DataList{}
			   opts := &client.ListOptions{}
			   err = c.List(context.TODO(), &dataObjects, opts)
			   Expect(err).NotTo(HaveOccurred())
			   Expect(len(dataObjects.Items)).To(Equal(1))
		           if tc.expectTemplateReference {
                            Expect(dataObjects.Items[0].Spec.TemplateReference).To(Equal(tc.template1.Name))
			   } else {
			      Expect(dataObjects.Items[0].Spec.TemplateReference).ToNot(Equal(tc.template1.Name))
			   }
```

Reply:
Didnt get your point @furkatgofurov7

Reply:
Github UI made it look ugly, point was in your tests maybe we could take lines 476-480 outside of the if condition since it is repeated in both if/else.

Reply:
Changed. Thanks. I have changed the test cases a little bit also. PTAL again